### PR TITLE
[IMP] project: Generic improvements on projects

### PR DESCRIPTION
**Pull request by bastvdn**
Description of the issue/feature this PR addresses:

- project.task > search parent task > if possible, add a default filter on the current project-->pending
- my tasks list view: if possible display the personal stage instead of the stage for private tasks
- project.task form view > sub-tasks notebook > add a line > apply a default filter on 'open tasks'
- indicate '(copy)' in the name of the stage when duplicating it

Current behavior before PR:

Desired behavior after PR is merged:

Task-3024152

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/project/views/project_views.xml
Comment (MissingNoShiny) [inline addons/project/views/project_views.xml:1527]: This is good when there are multiple projects in the context, but if we are viewing the tasks of a single project, we want to be able to edit this value.
To achieve this, I think we will need to create a view inheriting this one, where the `stage_id` field will be replaced by `stage_display`, like this: 
```xml
<record id="new_view_inheriting_the_list_view" model="ir.ui.view">
    <field name="name">Name</field>
    <field name="model">project.task</field>
    <field name="inherit_id" ref="view_task_tree2"/>
    <field name="mode">primary</field>
    <field name="arch" type="xml">
        <field name="stage_id" position="replace">
            <field name="stage_display" optional="show" readonly="1"/>
        </field>
    </field>
</record>
```
We can then use this view in the `action_view_all_task` action by setting it here:
```diff
diff --git a/addons/project/views/project_views.xml b/addons/project/views/project_views.xml
index 8d3100ced0b..6a942fb4b70 100644
--- a/addons/project/views/project_views.xml
+++ b/addons/project/views/project_views.xml
@@ -1681,6 +1681,7 @@
         <record id="open_view_all_task_list_tree" model="ir.actions.act_window.view">
             <field name="sequence" eval="20"/>
             <field name="view_mode">tree</field>
+            <field name="view_id" ref="new_view_inheriting_the_list_view"/>
             <field name="act_window_id" ref="action_view_all_task"/>
         </record>
         <record id="open_view_all_task_list_calendar" model="ir.actions.act_window.view">
```
Let me know if you need help for this :slightly_smiling_face:

**Review by MissingNoShiny — COMMENTED** [review]
Hello, thanks for your work :slightly_smiling_face: 
I have left a few comments.

Could you please read [the guidelines regarding commits and PRs](https://www.odoo.com/documentation/16.0/contributing/development/coding_guidelines.html#git)?
In particular, your commit messages should follow this template:
```
[IMP] module (not capitalized): short sentence that can be prefixed by 'This commit will ...'

Long description, rationale if relevant

Related PRs (if you have multiple PRs) and Task ID (in your case, "Task-3024152")
```

If you need to correct things you did in a commit, either squash/amend it directly with the original commit, or create a fixup commit that we will squash before the merge (this is especially useful if your task has already been reviewed, so that the reviewer can see what you changed since the last review).

When opening a PR in community, you can get rid of the message that is there by default, it's there for community contributor. Once again, it should reference related PRs and contain the id of the task.

Also, I think there might be a configuration issue with your git, because I see that some of your commits have a different author and committer.

Let me know if you have any question :slightly_smiling_face:

File: addons/project/views/project_views.xml
Comment (bastvdn) [reply addons/project/views/project_views.xml]: I didn't know you could replace fields using the xml, i thought it was only possible using xpath on kanban.
Thanks a lot for the tips, it seems to be working,not 100% if it's okay :) tell me if anything is wrong!

**Comment by bastvdn** [comment]
Thanks a lot for the review, very insightful!!

i followed the template for my fixup commit, i hope it's okay.
Concerning the configuration issue of my git, for the first commit i've done, i used my personal github account without setting up my global github env, but it should be okay know, i'm passing the CLA test on the PR :)
I tried to fix everything you commented, i know this PR isn't big at all but it took me a bit of time before getting everything in check, thanks again for the review and do not hesitate to tell me if anything isn't perfect yet!!

File: addons/project/models/project.py
Comment (MissingNoShiny) [inline addons/project/models/project.py:1226]:
```suggestion
    stage_display = fields.Char(compute='_compute_stage_display', string='Stage Display')
```
The `stage_id` field already uses "Stage" as a label, and we shouldn't have two fields with the same label

File: addons/project/views/project_views.xml
Comment (MissingNoShiny) [inline addons/project/views/project_views.xml:1679]:
```suggestion
                    <field name="stage_display" optional="show"/>
```
The readonly is actually not necessary, since the field is computed, and therefore already readonly (my bad)

**Review by MissingNoShiny — COMMENTED** [review]
Hello, I left two more comments, after that LGTM :slightly_smiling_face: 

I see that you did a fixup commit. It is not necessary to give them a commit message, as it will get discarded anyway when the commit will be squashed.

It would be good to not have multiple commits for the same generic improvement, so, for simplicity's sake, I would advise to squash all of your changes in a single commit and write a message for all the changes at once.
Usually, for generic imps, I like to have them separated in different commits, but as they are already mixed up here, it will be easier for you to group everything in a single commit.

File: addons/project/views/project_views.xml
Comment (bastvdn) [reply addons/project/views/project_views.xml]: Done :)

File: addons/project/models/project.py
Comment (bastvdn) [reply addons/project/models/project.py]: Done :)

File: addons/project/views/project_views.xml
Comment (MissingNoShiny) [inline addons/project/views/project_views.xml:1673]: Just a small last request, this view should be called something different than "Name". :innocent: 
You can just copy the id and replace the `_` with `.` if you don't have any inspiration
```suggestion
            <field name="name">open.view.my.tasks.list.view</field>
```

File: addons/project/models/project.py
Comment (IT-Ideas) [inline addons/project/models/project.py:142]: Could you rephrase the commit like this?
```
[IMP] project: improve user experience

This commit introduces the following improvements:

- In `My Tasks` list view: the personal stage is displayed instead of the stage for private tasks
- When copying a stage, the new stage name is appended with `(copy)`
- A default filter 'open task' is addd in the subtask notebook search
- A default filter on the project is set on the search of the `parent_id` task field

task-3024152
```
The commit tiles should start with a verb.
The prior / after this commit is rather used when changing a business behavior and when it seems really important to remind on the previous behavior. I would say that here it was almost the repeating what has been introduced. That's why I would rather focus on what is introduced.

**Review by IT-Ideas — COMMENTED** [review]
Just a little change on the commit message.
Thanks for your work 🙂

File: addons/project/models/project.py
Comment (bastvdn) [reply addons/project/models/project.py:142]: Done, thanks for the precision about the commit message :)

**Review by IT-Ideas — APPROVED** [review]
Thanks 🙂 
@robodoo r+

**Pull request closed without merging**
